include/oneapi/dpl/experimental/kt/two_pass_scan.h

```diff
@@ -267,13 +318,31 @@ two_pass_scan(sycl::queue q, _InRng&& __in_rng, _OutRng&& __out_rng,
                     //         on WG T-local carries
                     //            0: T0 carry, 1: T0 + T1 carry, 2: T0 + T1 + T2 carry, ...
                     //           S: sum(T0 carry...TS carry)
-                    constexpr std::uint8_t iters = num_sub_groups_local / VL;
+                    std::uint8_t iters = oneapi::dpl::__internal::__dpl_ceiling_div(num_sub_groups_local, VL);
                     auto csrc = g * num_sub_groups_local;
-                    _ONEDPL_PRAGMA_UNROLL
-                    for (std::uint8_t i = 0; i < iters; i++)
+                    if (is_full_carry_scanner)
+                    {
+                        _ONEDPL_PRAGMA_UNROLL
+                        for (std::uint8_t i = 0; i < iters; i++)
+                        {
+                            sub_group_partials[i * VL + sub_group_local_id] =
+                                tmp_storage[csrc + i * VL + sub_group_local_id];
+                        }
+                    }
+                    else
                     {
-                        sub_group_partials[i * VL + sub_group_local_id] =
-                            tmp_storage[csrc + i * VL + sub_group_local_id];
+                        std::uint8_t i = 0;
+                        _ONEDPL_PRAGMA_UNROLL
```

Comment:
I don't think we are going to get anything out of these unrolls anymore because we have removed the constexpr calculation of `iters`.  It probably doesn't hurt us, but it may be misleading.

Reply:
I have removed the unrolls here and in a few other places where our loops can't be unrolled.